packages/store/addon/-private/system/store/finders.js

```diff
@@ -302,7 +303,8 @@ export function _findHasMany(adapter, store, internalModel, link, relationship,
 export function _findBelongsTo(adapter, store, internalModel, link, relationship, options) {
   let snapshot = internalModel.createSnapshot(options);
   let modelClass = store.modelFor(relationship.type);
-  let promise = adapter.findBelongsTo(store, snapshot, link, relationship);
+  let relatedLink = !link || typeof link === 'string' ? link : link.href;
```

Comment:
if link is null doesn't this make relatedLink === true?

Reply:
no, `!link` would result in the value being `link` so `!null` (true) results in the value being `null`. This is the value of the ternary operator and is equivalent to

```ts
(!link || typeof link === 'string') ? link : link.href
```

![image](https://user-images.githubusercontent.com/650309/67246374-9b3bc700-f45e-11e9-97be-6c4d713c41fc.png)

Reply:
Oh god, okay, please add some parens :)

Reply:
(also fwiw I authored with the parens for clarity but sadly prettier removes)